### DISCUSSION TITLE
Making diacritics work

### DISCUSSION THREAD

**Discussion by iandoug** [binarybottle/engram-es#23] [Ideas]
Hi

Is anybody looking at how to make the DiacriticKey actually work on the various OSs?

I'll need to figure out how to support this concept in KLA... dead keys are totally not supported, let alone Magic dead keys....

**Comment by iandoug** [comment]
Looks like XKB does not support Magic Diacritic keys. Don't know about the other OSs.

https://unix.stackexchange.com/questions/407154/create-own-dead-key
http://linux.lsdev.sil.org/wiki/index.php/Building_an_XKB_Keyboard#Dead_Keys_and_Diacritics

It's probably doable with KMonad or AutoHotKey.

**Comment by Lobo-Feroz** [comment]
Well, I don't know how one would do a "Magic" Diacritic Key that would change the diacritics based on the next character.

But the mostly used diacritic in spanish can be extended to do more diacritics like the proposal for N and C.

This is how it looks in MS Keyboard Layout Creator:

![dead-keys](https://user-images.githubusercontent.com/10242383/129410949-8e6ce1bc-ac1b-4781-98b0-77457c264e60.png)

You simply add more combinations to the dead key. 

I suppose it will be similar in ubuntu, if you want I can research it next.

**Comment by binarybottle** [comment]
My preference would be to have any sequence or combination of keys be (1) as ergonomically comfortable as possible, (2) work on ansi as well as iso, (3) work on all operating systems, and (4) preferably work on keyman.

**Reply by iandoug** [reply]
I don't think xkb supports that approach. I would do the equivalent via the Compose key, but that is not "typing flow friendly".

eg Compose+c+, = ç ... although I have that on AltGr+c as well.
One could define Compose+c+' = ç if one so wished, but , is better visually.

Mmm... I stand corrected ... looks like it is directly possible:
https://xkbcommon.org/doc/current/group__compose.html
<dead_acute> <a>         : "á"   aacute # LATIN SMALL LETTER A WITH ACUTE

Now it's just Mac to sort out.
